core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java

```diff
@@ -169,7 +175,19 @@ public void prepareRun(BatchSourceContext context) throws Exception {
 
     conf.set(CUTOFF_READ_TIME, dateFormat.format(prevHour));
     FileInputFormat.setInputPathFilter(job, BatchFileFilter.class);
-    FileInputFormat.addInputPath(job, new Path(config.path));
+
+    if (!fs.exists(new Path(config.path)) && config.ignoreNonExistingFolders) {
+      path = new Path(fs.getWorkingDirectory() + "/tmp");
+      fs.mkdirs(path);
+      conf.set(INPUT_NAME_CONFIG, path.toUri().getPath());
+      FileInputFormat.addInputPath(job, path);
+      LOG.warn("File/Folder specified in path does not exists. Setting input path to " +
```

Comment:
Should you log the warning first, in case something fails when trying to create the /tmp dir?